pkg/controller/cleanup/metrics.go

```diff
@@ -20,104 +20,40 @@ import (
 
 	"go.opencensus.io/stats"
 	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
 )
 
 const metricPrefix = observability.MetricRoot + "/cleanup"
 
 var (
-	mClaimAttempts                  = stats.Int64(metricPrefix+"/claim_attempts", "The number of attempts to claim the cleanup", stats.UnitDimensionless)
-	mClaimErrors                    = stats.Int64(metricPrefix+"/claim_errors", "The number of errors to claim the cleanup", stats.UnitDimensionless)
-	mPurgeVerificationCodesAttempts = stats.Int64(metricPrefix+"/purge_verification_codes_attempts", "The number of attempts to purge verification codes", stats.UnitDimensionless)
-	mPurgeVerificationCodesErrors   = stats.Int64(metricPrefix+"/purge_verification_codes_errors", "The number of attempts to purge verification codes", stats.UnitDimensionless)
-	mPurgeVerificationCodesPurged   = stats.Int64(metricPrefix+"/purge_verification_codes_purged", "The number of verification codes that were purged", stats.UnitDimensionless)
-
-	// Verification tokens
-	mPurgeVerificationTokensAttempts = stats.Int64(metricPrefix+"/purge_verification_tokens_attempts", "The number of attempts to purge verification tokens", stats.UnitDimensionless)
-	mPurgeVerificationTokensErrors   = stats.Int64(metricPrefix+"/purge_verification_tokens_errors", "The number of attempts to purge verification tokens", stats.UnitDimensionless)
-	mPurgeVerificationTokensPurged   = stats.Int64(metricPrefix+"/purge_verification_tokens_purged", "The number of verification tokens that were purged", stats.UnitDimensionless)
-
-	// Mobile apps
-	mPurgeMobileAppsAttempts = stats.Int64(metricPrefix+"/purge_mobile_apps_attempts", "The number of attempts to purge mobile apps", stats.UnitDimensionless)
-	mPurgeMobileAppsErrors   = stats.Int64(metricPrefix+"/purge_mobile_apps_errors", "The number of attempts to purge mobile apps", stats.UnitDimensionless)
-	mPurgeMobileAppsPurged   = stats.Int64(metricPrefix+"/purge_mobile_apps_purged", "The number of mobile apps that were purged", stats.UnitDimensionless)
+	mRequests      = stats.Int64(metricPrefix+"/requests", "The number of cleanup requests.", stats.UnitDimensionless)
+	mClaimRequests = stats.Int64(metricPrefix+"/claim_requests", "The number of cleanup claim requests.", stats.UnitDimensionless)
+)
 
-	// Audit entries
-	mPurgeAuditEntriesAttempts = stats.Int64(metricPrefix+"/purge_audit_entries_attempts", "The number of attempts to purge audit entries", stats.UnitDimensionless)
-	mPurgeAuditEntriesErrors   = stats.Int64(metricPrefix+"/purge_audit_entries_errors", "The number of attempts to purge audit entries", stats.UnitDimensionless)
-	mPurgeAuditEntriesPurged   = stats.Int64(metricPrefix+"/purge_audit_entries_purged", "The number of audit entries that were purged", stats.UnitDimensionless)
+var (
+	// itemTagKey indicating what type of items is cleaned up in this step.
+	// Potential values:
```

Comment:
I'm slightly worried about this approach as it wouldn't catch typo until the metric hit prod (e.g. `VERIFICATON_CODE`). Is there a way we can validate the allowed list somehow?

Reply:
We should start by alerting on the over all error ratio. From my day job experience, further breakdown to a particular error is usually not very useful.

We can set up a chart that breaks down to detailed errors, but that is used by the oncaller during alert troubleshooting.

FWIW the alert policy can also set up a "data point has been missing for x min" condition so we should be able to catch typos in the filter.